src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -203,19 +204,31 @@ def create_if_needed(self, app_name):
             )  # TODO use .info()
 
     def create(self):
-        self.location = validate_environment_location(self.cmd, self.location)
         register_provider_if_needed(self.cmd, LOG_ANALYTICS_RP)
-        env = create_managed_environment(
-            self.cmd,
-            self.name,
-            location=self.location,
-            resource_group_name=self.resource_group.name,
-            logs_key=self.logs_key,
-            logs_customer_id=self.logs_customer_id,
-            disable_warnings=True,
-        )
-        self.exists = True
-        return env
+
+        res_locations = list_environment_locations(self.cmd)
+        for loc in res_locations:
+            try:
+                env = create_managed_environment(
+                    self.cmd,
+                    self.name,
+                    location=self.location,
```

Comment:
Shouldn't this be location=loc?

Reply:
Looks like this code will keep trying the same location for every loc in res_locations, since we only assign loc to self.location if the env create is successful. Might make sense to keep location=self.location to honor user input, and have self.location = loc at the end of the for loop. So flow would be fail -> log failure -> iterate location.

Reply:
HIstorically, we honor user parameter. For instance, if user passes a location that has full environment quota, we do not try other locations, we simply show them the error message. Only if the user does not pass location do we want to auto-select location. I would reach out to @anthonychu for more details.

Reply:
I have updated the code to only create once when there is user-input location. When there is no location input, will iterate on res_locations, assign it to self.location first and try to create with selt.location

Thanks for reminding! Please check if this will still break the user experience